hubblestack/extmods/utils/s3.py

```diff
@@ -250,6 +250,9 @@ def query(key, keyid, method='GET', params=None, headers=None,
         return 'Saved to local file: {0}'.format(local_file)
 
     if result.status_code < 200 or result.status_code >= 300:
+        if err_code == 'SlowDown':
+            log.warning('%s, %s', err_code, err_msg)
```

Comment:
Currently during the main daemon loop the raised exception would generate the log (below) that is easy to search for.
Since this is to change, couple of questions: 
- How would an example log look like if a 503 is hit? Can we prepend a string like the one below that helps to identify that it's a matter of s3?
- Will the log end up being sent to Splunk ?
- Would it make sense to handle other error codes to avoid a crash on start up? For example, I think ServiceUnavailable, RequestTimeTooSkewed, RequestTimeout, OperationAborted, InternalError can be eventually recoverable or may be intermittent. (https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html)

Reply:
I can prefix the log with a string that will help us identify its an s3 issue. Currently, it would just say 'Slowdown, Please reduce your request rate' if that 503 error were to occur. In terms of being sent to Splunk, that would only be through CloudTrail & Object level logging on s3 or access level logging(ListObject api calls). Some of these errors would make sense to not raise an exception and handle that exception properly.